### DISCUSSION TITLE
Edit mode renders KaTeX math in markdown table differently

### DISCUSSION THREAD

**Discussion by matt035343** [requarks/wiki#5842] [Error / Bug Report]
Version: 2.5.291

I spotted a bug related to KaTeX math in markdown tables. The bug can be reproduced with following code:
```markdown
| Math            | Description               |
|-----------------|:--------------------------|
| $\mathcal{S}$   | Math does not render      |
| $\mathcal{T}$   | Math does not render      |
| $\mathcal{U}_~$ | Workaround, now rendering |
```

Strangely, if I append `_~` to the above math, the math renders again.

This is what I see when I preview the page with math in edit mode:
![image](https://user-images.githubusercontent.com/3768287/201690747-c132e165-0c51-4802-9e4d-d6c784de9e97.png)

This is what I see when I visit my page without edit mode:
![image](https://user-images.githubusercontent.com/3768287/201691028-081b56ce-4863-46cc-9948-acaa532065e1.png)

Any advice is welcome.

**Comment by NGPixel** [comment]
Try to disable HTML Sanitization under Admin > Rendering > HTML > Security, and re-save your page?

**Reply by matt035343** [reply]
Hi, thanks for the suggestion. It does not seem to fix the issue. Any other ideas?

**Reply by matt035343** [reply]
@NGPixel Any updates on this?

**Reply by qurban** [reply]
Additional example...

![image](https://user-images.githubusercontent.com/515220/205749993-61539b72-2ea1-4c84-aa29-d47087df6d4c.png)

renders as

hello:

in the editor

but renders as

\text

on the actual page after saving.

However, if I add any kind of actually parsed code to the block, for example:

![image](https://user-images.githubusercontent.com/515220/205750046-4dde0030-3d1f-4a74-a5f9-f3ecf9d066c4.png)

then it renders as

hello:

in the editor

and as

hello:

in the saved page.

\mathcal seems conceptually similar to \text in that they both render unprocessed text as opposed to processed text.

Hope this helps.

**Reply by matt035343** [reply]
@NGPixel Do you know if this issue has a planned fix? I think we are many users experiencing this issue.

**Comment by qurban** [comment]
I think I've narrowed it down even more.

If the last character in a katex code block, before the final $, is a } then the block will render properly in the preview but not render properly in the final page.

Sample Code
![image](https://user-images.githubusercontent.com/515220/205975923-d256ab0c-2710-4784-b747-cfafd0e0bf05.png)

Rendering in Preview
![image](https://user-images.githubusercontent.com/515220/205976009-a30ffbcd-3c3d-45a3-95bf-39e08491b897.png)

Rendering in Final Page
![image](https://user-images.githubusercontent.com/515220/205976063-5f86f901-9e02-4931-9f8c-d4a000300df0.png)

**Reply by qurban** [reply]
So minor update... if the katex block ends in }$ and then is followed by a carriage return or space then it does not render properly in the final page.

The same block is inline with other text, then it works.

![image](https://user-images.githubusercontent.com/515220/205979681-993e1b25-932b-415a-b5d0-af37c1f7a851.png)

![image](https://user-images.githubusercontent.com/515220/205979701-03e9051e-1f85-4955-a10e-4d90faea43d2.png)

**Comment by jim-kirisame** [comment]
This problem is same as issue#1462. 

The formula will render failed when "}" at the end and no other non-spacing character after the "$". You could add some extra character before or after "$"

```
render fail: 
$\frac{1}{2}$

works well:
$\frac{1}{2}\\$
$\frac{1}{2}$.
```

**Reply by qurban** [reply]
Thanks!
